tools/helpers.cmake

```diff
@@ -204,30 +204,32 @@ function(GenProofsBFTarget target_name target_file pbf_path pbf_target prunes de
     )
 endfunction(GenProofsBFTarget)
 
-macro(cfg_str_add_disabled cfg_str name)
+macro(cfg_str_add_disabled cfg_str cfg_json name)
     # include in liner files can't use double slash comments
     list(APPEND ${cfg_str} "/* disabled: CONFIG_${name} */")
+    string(JSON ${cfg_json} SET ${${cfg_json}} "${name}" "null")
```

Comment:
It's really sad that JSON does not allow comments, because I consider this something that should not show up in the "code" to avoid bloating and confusion. Is there any gain adding `null` explicitly when you could also handle this when parsing? My long intention when making this visible as comment in the header was having a way to see whgat potential configuration quirks exists (like setting some ARM values when not targeting ARM at all)  that should be looked into one day.

Reply:
When developing it, it was useful to know that all of the data was handled, but it doesn't have to be like this long term, no. Most parsers don't let you determine the difference between something in json being null vs not existing. Given that humans don't need to look at this file (in fact, them looking at it only temps them to change it, which defeats the goal), I think not having comments is fine.

Reply:
I think we do need to output unset options for consumers of this file. Otherwise you have to add explicit logic for options that don't exist in the consumer, and it'd be extremely easy to introduce bugs by misspelling option names. I.e. it'd be good if a missing option could be an error. Similarly, an option that shows up but is not expected could be a warning that something is not right.

Reply:
Would it be worthwhile to use a markup that allows for comments instead? I think yaml seemed to be met with interest/approval during the hangouts.

Reply:
@lsf37 made the remark that it is intended for machine processing in the first place. Given people look at this still to understand what is happening under the hood, it has some benefit to allow comments. Making  debuggigng and prototyping easier is another use case.

Reply:
1. This is intended for output, not input, therefore, users aren't supposed to change it. Having comments would only encourage people to read it and that would only encourage them to change it. If CBOR were as easy to produce as JSON, I'd just as soon use that so when people opened it, it was just as useful as opening kernel.elf, cause realistically, the need to have comments in both files are equivalent.